RENO-1463: Add urlType option to support absolute or relative links

misselbeck opened the pull request:
[Related Jira ticket](https://jira.nypl.org/browse/RENO-1463)

**This PR does the following:**
- Changes footer links to be stored as absolute urls
- Adds a new propType called `urlType`, which allows "absolute" to be passed into the Footer component
- Adds a function to make the links relative if the "absolute" prop is not passed in.

This work allows apps such as `sfr-bookfinder-front-end` to configure the footer component to use absolute links, since it lives at a different domain.

The rest of the apps will continue to get relative links by default, as that is more helpful during the QA process.

The approach is identical to the way the `dgx-header-component` handles it.

**Testing:**
- I tested this branch against a local version of `sfr-bookfinder-front-end` and it worked as expected.
- You can also test locally by adding the prop to the Footer component in app.jsx:

`<Footer id="footer" className="footer" urlType="absolute" />`

EdwinGuzman commented:
This looks good but can you write tests for the regex function and also update the readme to include an example of how to use the new prop? Similar to what's in the PR description.

EdwinGuzman commented:
I just noticed there are no tests or test running in this app. That's a problem... so maybe instead of tests can you write more comments explaining what the code does?

misselbeck commented:
I made some adjustments, as suggested.

- Moved the regex function into a utils file for reusability and better commenting
- Updated the README with instructions and example of the urlType prop
